src/fuzzycmeans.jl

```diff
@@ -0,0 +1,112 @@
+# Fuzzy C means algorithm
+
+## Interface
+
+immutable FuzzyCMeansResult{T<:AbstractFloat} <: ClusteringResult
+    centers::Matrix{T}          # cluster centers (d x C)
+    weights::Matrix{T}          # assigned weights (n x C)
```

Comment:
If you make `centers` take on the type of the input `data`, then it's possible that `weights` will have a different type (e.g., in k-means, the centers have a parameterized type `T`, but the weight vector is a `Float64`).  How about doing the same thing here?